### PR TITLE
Add OWASP Zap scan

### LINKED ISSUE #46: dynamic security scan
Task broken out of #33 

Scope:

- [x] OWASP Zap scan running locally
- [x] OWASP Zap scan running in actions

### PR DESCRIPTION
**Pull request by rahearn**
closes #46

### CODE REVIEW

File: .github/actions/run-server/action.yml [hunk 1]
```diff
@@ -7,8 +7,7 @@ inputs:
     default: ci
   database_url:
     description: DATABASE_URL to set
-    required: false
-    default: postgres://pangolin:postgres@localhost:5432/pangolin_test
+    required: true
```
Comment (rahearn) [inline .github/actions/run-server/action.yml:10]: In the spirit of only having to update the url string in one place, requiring this to be passed in from the setup action outputs.

File: .github/actions/setup-languages/action.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Set up languages
```
Comment (rahearn) [inline .github/actions/setup-languages/action.yml:1]: Replaced the `setup-database` script with this one. Now each workflow only has to call one setup action. `setup-project` if a db is needed. `setup-languages` if no db is needed.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# EditorConfig is awesome: https://EditorConfig.org
```
Comment (hartsick) [inline .editorconfig:1]: ooooh this is so cool! didn't know something like this existed

File: .github/actions/run-server/action.yml (hunk 1 above)
Comment (hartsick) [reply .github/actions/run-server/action.yml:10]: 🎉

File: .github/actions/setup-languages/action.yml (hunk 2 above)
Comment (hartsick) [reply .github/actions/setup-languages/action.yml:1]: love it

File: zap.conf [hunk 4]
```diff
@@ -0,0 +1,121 @@
+# zap-full-scan rule configuration file
```
Comment (hartsick) [inline zap.conf:1]: are these suggestions from somewhere (owasp?), and is there customization we've done that would be helpful to know about?

File: bin/pa11y-scan [hunk 5]
```diff
@@ -6,25 +6,5 @@
...
-yarn install --frozen-lockfile
-
-yarn run pa11y-ci
-exit_status=$?
-
-# shut down the server
-kill $server_pid
-
-exit $exit_status
+`dirname "$0"`/with-server "yarn run pa11y-ci"
```
Comment (hartsick) [inline bin/pa11y-scan:10]: this is great. do you have other uses of `with-server` in mind?

**Review by hartsick — APPROVED** [review]
🎉

File: zap.conf (hunk 4 above)
Comment (rahearn) [reply zap.conf:1]: Um, I grabbed them from the complaint tracker repo I worked on with OneOHS. I honestly don't remember how we ended up with this exact set of warn/fails

File: bin/pa11y-scan (hunk 5 above)
Comment (rahearn) [reply bin/pa11y-scan:10]: Not immediately, beyond pa11y and owasp.

**Pull request merged**